packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -119,18 +136,139 @@ function getResizeDirection(element: HTMLElement, direction: Direction) {
   return side;
 }
 
+function preventClick(event: MouseEvent) {
+  event.preventDefault();
+  event.stopImmediatePropagation();
+}
+
+/**
+ * Checker that returns a promise that resolves when the column virtualization
+ * is disabled.
+ */
+function useColumnVirtualizationDisabled(apiRef: React.MutableRefObject<GridPrivateApiPro>) {
+  const promise = React.useRef<ControllablePromise>();
+  const selector = () => gridVirtualizationColumnEnabledSelector(apiRef);
+  const value = useGridSelector(apiRef, selector);
+
+  React.useEffect(() => {
+    if (promise.current && value === false) {
+      promise.current.resolve();
+      promise.current = undefined;
+    }
+  });
+
+  const asyncCheck = () => {
+    if (!promise.current) {
+      if (selector() === false) {
+        return Promise.resolve();
+      }
+      promise.current = createControllablePromise();
+    }
+    return promise.current;
+  };
+
+  return asyncCheck;
+}
+
+/**
+ * Basic statistical outlier detection, checks if the value is `F * IQR` away from
+ * the Q1 and Q3 boundaries. IQR: interquartile range.
+ */
+function excludeOutliers(inputValues: number[], factor: number) {
+  if (inputValues.length < 4) {
+    return inputValues;
+  }
+
+  const values = inputValues.slice();
+  values.sort((a, b) => a - b);
+
+  const q1 = values[Math.floor(values.length * 0.25)];
+  const q3 = values[Math.floor(values.length * 0.75) - 1];
+  const iqr = q3 - q1;
+
+  // We make a small adjustment if `iqr < 5` for the cases where the IQR is
+  // very small (e.g. zero) due to very close by values in the input data.
+  // Otherwise, with an IQR of `0`, anything outside that would be considered
+  // an outlier, but it makes more sense visually to allow for this 5px variance
+  // rather than showing a cropped cell.
+  const deviation = iqr < 5 ? 5 : iqr * factor;
+
+  return values.filter((v) => v > q1 - deviation && v < q3 + deviation);
+}
```

Comment:
I have some doubts about the UX outcome of this. Did we consider excluding rows if their width is larger than Q3 + a max outlier distance variable in px?

Reply:
An absolute value isn't adaptive enough for outliers detection in the general case, but it might work for this particular case due to the widths being pixel values. But I'm not totally convinced, most of the time outliers detection with the standard formula works for the huge majority of use cases. And the thing is, if users are at the point where they're configuring pixels variance values, then they already have a good idea of the approximate size of their cells, and they'd probably just set a width directly.

I think this is good enough for now, and if we ever have a request for that type of configuration we can trivially add an `outliersDeviation` option that would supersede `outliersFactor` if present. It would be non-breaking so it can be released quickly if there is a need for it.

Reply:
As a product designer, I suspect that the smaller the Q3 width is, the more space there is on the screen, the biggest the breakpoint can be tolerated, since there is more free space on the screen. And the bigger the Q3 width is, the least space there is on the screen, the smaller the breakpoint the better.

So do I want the breakpoint to be at Q3 width + `200px`, or 2*Q3 - Q1 width * `1.5`? I think Q3 width + `200px` makes more sense.

As a side node, I don't see why Q1 is in the formula. We have an optimization problem here, using the least amount of column width while seeing the maximum amount of information. So Q1, I don't get it.

Reply:
> As a side note, I don't understand why Q1 is in the formula. We have an optimization problem here, using the least amount of column width while seeing the maximum amount of information

I used the standard stats outlier detection formula: https://www.statisticshowto.com/statistics-basics/find-outliers. It's used for a very wide range of applications and usually "just works".

I'm not sure I'd trust an absolute number. Let's say we take `Q3 + 200px`, what if most of the widths are in the `[50, 60]` range? A cell with a width of `150px` is 3 times that, and wouldn't be detected as an outlier. The IQR (and by extension Q1, which is in there) is useful because it makes it more adaptive to the data.

Reply:
>  what if most of the widths are in the [50, 60] range? A cell with a width of 150px is 3 times that, and wouldn't be detected as an outlier.

I have found an issue related to this: https://github.com/mui/mui-x/issues/10456. IMHO this shouldn't be considered an outlier. This matches with the default behavior of Excel and Google Sheets which resizes including outliers.